src/webapp/azext_webapp/custom.py

```diff
@@ -28,6 +29,89 @@ def _ping_scm_site(cmd, resource_group, name):
+    scm_url = _get_scm_url(cmd, resource_group_name, name, slot)
+    start_scan_url = scm_url + '/api/scan/start/' + timeout
+
+    import urllib3
+    authorization = urllib3.util.make_headers(basic_auth='{0}:{1}'.format(user_name, password))
+    headers = authorization
+    headers['content-type'] = 'application/octet-stream'
+
+    response = requests.get(start_scan_url, headers=authorization)
+    return response.json()
+
+
+def get_scan_result(cmd, resource_group_name, name, scan_id, slot=None):
+    webapp = show_webapp(cmd, resource_group_name, name, slot)
+    is_linux = webapp.reserved
+    if not is_linux:
+        raise CLIError("Only Linux App Service Plans supported, Found a Windows App Service Plan")
+
+    import requests
+    user_name, password = _get_site_credential(cmd.cli_ctx, resource_group_name, name, slot)
+    scm_url = _get_scm_url(cmd, resource_group_name, name, slot)
+    scan_result_url = scm_url + '/api/scan/' + scan_id + '/result'
+
+    import urllib3
+    authorization = urllib3.util.make_headers(basic_auth='{0}:{1}'.format(user_name, password))
+    headers = authorization
+    headers['content-type'] = 'application/octet-stream'
+
+    response = requests.get(scan_result_url, headers=authorization)
+
+    return response.json()
+
+
+def track_scan(cmd, resource_group_name, name, scan_id, slot=None):
+    webapp = show_webapp(cmd, resource_group_name, name, slot)
+    is_linux = webapp.reserved
+    if not is_linux:
+        raise CLIError("Only Linux App Service Plans supported, Found a Windows App Service Plan")
+
+    import requests
+    user_name, password = _get_site_credential(cmd.cli_ctx, resource_group_name, name, slot)
+    scm_url = _get_scm_url(cmd, resource_group_name, name, slot)
+    scan_result_url = scm_url + '/api/scan/' + scan_id + '/track'
+
+    import urllib3
+    authorization = urllib3.util.make_headers(basic_auth='{0}:{1}'.format(user_name, password))
+    headers = authorization
+    headers['content-type'] = 'application/octet-stream'
+
+    response = requests.get(scan_result_url, headers=authorization)
+
+    return response.json()
+
+
+def get_all_scan_result(cmd, resource_group_name, name, slot=None):
+    webapp = show_webapp(cmd, resource_group_name, name, slot)
+    is_linux = webapp.reserved
+    if not is_linux:
+        raise CLIError("Only Linux App Service Plans supported, Found a Windows App Service Plan")
+
+    import requests
+    user_name, password = _get_site_credential(cmd.cli_ctx, resource_group_name, name, slot)
+    scm_url = _get_scm_url(cmd, resource_group_name, name, slot)
+    scan_result_url = scm_url + '/api/scan/results'
+
+    import urllib3
+    authorization = urllib3.util.make_headers(basic_auth='{0}:{1}'.format(user_name, password))
+    headers = authorization
+    headers['content-type'] = 'application/octet-stream'
+
+    response = requests.get(scan_result_url, headers=authorization)
+
+    return response.json()
```

Comment:
Why are these raw API calls instead of SDK calls?

Reply:
We plan to add this feature in Preview, and it is still under development. That is why these are direct API calls, and was added to CLI extensions. We plan to add SDK calls later.